pkg/controller/cluster/controller.go

```diff
@@ -23,33 +23,90 @@ import (
 
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
 
-	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"github.com/golang/glog"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util"
 )
 
 // +controller:group=cluster,version=v1alpha1,kind=Cluster,resource=clusters
 type ClusterControllerImpl struct {
 	builders.DefaultControllerFns
 
-	// lister indexes properties about Cluster
-	lister listers.ClusterLister
+	actuator Actuator
+
+	// clusterLister holds a lister that knows how to list Cluster from a cache
+	clusterLister listers.ClusterLister
+
+	kubernetesClientSet *kubernetes.Clientset
+	clientSet           clientset.Interface
+	clusterClient       v1alpha1.ClusterInterface
 }
 
 // Init initializes the controller and is called by the generated code
 // Register watches for additional resource types here.
-func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments) {
-	// Use the lister for indexing clusters labels
-	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters().Lister()
+func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments, actuator Actuator) {
+	cInformer := arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters()
+
+	c.clusterLister = cInformer.Lister()
+
+
+	c.actuator = actuator
+
+	cs, err := clientset.NewForConfig(arguments.GetRestConfig())
+	if err != nil {
+		glog.Fatalf("error creating cluster client: %v", err)
+	}
+	c.clientSet = cs
+	c.kubernetesClientSet = arguments.GetSharedInformers().KubernetesClientSet
+
+	// Create cluster actuator.
+	// TODO: Assume default namespace for now. Maybe a separate a controller per namespace?
+	c.clusterClient = cs.ClusterV1alpha1().Clusters(corev1.NamespaceDefault)
+	c.actuator = actuator
 }
 
-// Reconcile handles enqueued messages
-func (c *ClusterControllerImpl) Reconcile(u *v1alpha1.Cluster) error {
-	// Implement controller logic here
-	log.Printf("Running reconcile Cluster for %s\n", u.Name)
+// Reconcile handles enqueued messages. The delete will be handled by finalizer.
+func (c *ClusterControllerImpl) Reconcile(cluster *clusterv1.Cluster) error {
+	name := cluster.Name
+	log.Printf("Running reconcile Cluster for %s\n", name)
```

Comment:
consistency: we are using glog on line 82 of this function (and elsewhere in this file). Please switch this line to use glog.